sdk/iptables/iptables.go

```diff
@@ -69,6 +69,11 @@ type Config struct {
 
 	// IptablesProvider is the Provider that will apply iptables rules.
 	IptablesProvider Provider
+
+	// AddAdditionalRulesFn can be implemented by the caller to
+	// add environment specific rules (like ECS) that needs to
+	// be executed for traffic redirection to work properly.
+	AddAdditionalRulesFn func(iptablesProvider Provider)
```

Comment:
~ A couple of minor suggestions, if we keep this func as-is:
- Doc that only `AddRule` and (optionally) `Rules` should be called by the provided function - or split out those funcs to their own `RuleAdder` interface, embed that interface in `Provider`, and pass `RuleAdder` to the function here instead
- Doc that this function is called from `Setup`, after all the build-in rule additions (just for clarity)

Something I noticed that seems worth calling out: As things stand, the `Setup` function is essential, but because it calls `ApplyRules` at the end, there's currently no safe way to add rules without also pre-configuring a provider rather than accepting the default. Given that constraint, this seems like a reasonable approach to me. The alternative I can imagine would be to provide a static list of additional rules to the config, which would also support serialization of those rules (e.g. injected by some optional config string) - with the tradeoff that you can't inspect existing rules when adding the custom ones, though I'm not sure that's the intent here.

Anywhere we're in control of the config struct and calling `Setup`, if the one-time constraint didn't exist, it seems like we could just call `AddRule` and `ApplyRules` directly instead of using a hook. In the future, we might want to consider making `ApplyRules` clear the list of applied commands s.t. it's safe to call `AddRule` and `ApplyRules` a second time, or provide a `Setup` alternative that doesn't apply rules automatically.